schema/samcli.json

```diff
@@ -143,9 +143,175 @@
               "type": "object",
-              "properties": {}
+              "properties": {
+                "config_env": {
+                  "title": "config_env",
+                  "type": "string",
+                  "description": "Environment name specifying default parameter values in the configuration file.",
+                  "default": "default"
+                },
+                "config_file": {
+                  "title": "config_file",
+                  "type": "string",
+                  "description": "Configuration file containing default parameter values.",
+                  "default": "samconfig.toml"
+                },
+                "guided": {
+                  "title": "guided",
+                  "type": "boolean",
+                  "description": "Specify this flag to allow SAM CLI to guide you through the deployment using guided prompts."
+                },
+                "template_file": {
+                  "title": "template_file",
+                  "type": "string",
+                  "description": "AWS SAM template which references built artifacts for resources in the template. (if applicable)",
+                  "default": "template.[yaml|yml|json]"
+                },
+                "no_execute_changeset": {
+                  "title": "no_execute_changeset",
+                  "type": "boolean",
+                  "description": "Indicates whether to execute the change set. Specify this flag to view stack changes before executing the change set."
+                },
+                "fail_on_empty_changeset": {
+                  "title": "fail_on_empty_changeset",
+                  "type": "boolean",
+                  "description": "Specify whether AWS SAM CLI should return a non-zero exit code if there are no changes to be made to the stack. Defaults to a non-zero exit code.",
+                  "default": true
+                },
+                "confirm_changeset": {
+                  "title": "confirm_changeset",
+                  "type": "boolean",
+                  "description": "Prompt to confirm if the computed changeset is to be deployed by SAM CLI."
+                },
+                "disable_rollback": {
+                  "title": "disable_rollback",
+                  "type": "boolean",
+                  "description": "Preserves the state of previously provisioned resources when an operation fails."
+                },
+                "on_failure": {
```

Comment:
Some of the parameters are incompatible with each other, should we hint towards this at the config level? (we might have already had this conversation before too)

Reply:
We did have this conversation, yea. I think leaving it as is should be fine, both because if the user tries to run the command with those params anyway, the CLI will throw them an error saying they can't, and they can resolve it on their own; and also because that logic is more tied to the logic of the command itself, rather than being something we can find in the click option.